src/site/_includes/components/Widget.js

```diff
@@ -1,27 +1,72 @@
-const patterns = require('../../_data/patterns').patterns;
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const path = require('path');
+const {findByUrl} = require('../../_filters/find-by-url');
 
 /**
  * @fileoverview A component to display an interactive demo,
- * specified as part of a code pattern in /patterns/ directory,
  * as an iframe embed.
  */
 
 /**
- * @param {string} patternId Id of the Code Pattern to be displayed.
+ * @param {string} demoUrl Url of the demo to be displayed. It can be
+ *   an external url, or a deep url on web.dev, .e.g. /demos/my-demo-name.
  * @param {string} height Optional height of the widget with units, e.g. 100px.
  */
-module.exports = (patternId, height) => {
-  const pattern = patterns[patternId];
-  if (!pattern) {
-    return '';
+module.exports = function (demoUrl, height) {
+  let url;
+  try {
+    // Check if it's external url.
+    const externalUrl = new URL(demoUrl);
+    url = externalUrl.href;
+  } catch (e) {
+    // Otherwise, check if internal url exists.
+    url = path.join('/', demoUrl, '/');
+    if (!findByUrl(url)) {
+      return '';
```

Comment:
Been wondering this with other shortcodes too: for amp.dev we rendered an aside in those cases during development saying something like "You're shortcode is not showing anything because of reasons". A simple `console.warn` would also work for me, just so that authors know they gave input that doesn't work for that shortcode.